addons/mail/models/res_users.py

```diff
@@ -49,7 +49,19 @@ def create(self, vals_list):
         return users
 
     def write(self, vals):
+        user_portal_access_dict = {}
+        for user in self:
+            user_portal_access_dict[user.id] = user.has_group('base.group_portal')
         write_res = super(Users, self).write(vals)
+        for user in self:
+            user_has_group = user.has_group('base.group_portal')
+            msg = ''
+            if user_has_group and not user_portal_access_dict[user.id]:
+                msg = 'Portal Access Granted'
+            elif not user_has_group and user_portal_access_dict[user.id]:
+                msg = 'Portal Access Revoked'
+            if msg:
+                user.partner_id.message_post(body=msg)
```

Comment:
I would post directly and not in a condition after, also you should translate the string and ensure that we do not send an email (the note subtype will set is_internal to True on the message which is good).

```python
for user in self:
    user_has_group = user.has_group('base.group_portal')
    if user_has_group and not user_portal_access_dict[user.id]:
        user.partner_id.message_post(body=_('Portal Access Granted'), message_type='comment', subtype_xmlid='mail.mt_note')
    elif not user_has_group and user_portal_access_dict[user.id]:
        user.partner_id.message_post(body=_('Portal Access Revoked'), message_type='comment', subtype_xmlid='mail.mt_note')
```

Reply:
Hello @std-odoo 

Changes done.
and also we did changes in create method. because, when we grant access from the portal wizard, it will create the user having portal access.
so in that case we are not getting log information.

Thanks